### PR TITLE
Add notifications table to the admin app

### PR DESCRIPTION
**Pull request by garrettqmartin8**
## Description

In order to more easily debug which notification emails are sent to whom, we needed a way to view them in the admin app.

## Reviewer Notes

You may notice that I've introduced a more generic service object (using `interface{}` as its secret sauce) that allows fetching lists of records for any table. I've made use of it here for fetching notifications and, for the sake of example, replaced the office user list fetcher. Clear, consistent patterns have emerged across our service objects and I'd like to save on some boilerplate, so I'll be trying to trim more fat in the same way in forthcoming PRs. Let me know what you think!

## Setup

- `make server_run`
- `make client_run`
- Log into the admin app and click on `Notifcations` in the sidebar

## Code Review Verification Steps

* [x] Request review from a member of a different team.
* [ ] Have the Pivotal acceptance criteria been met for this change?

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-566) for this change

### CODE REVIEW

File: pkg/services/fetch.go [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package services
+
+// ListFetcher is the exported interface for fetching multiple records
+//go:generate mockery -name ListFetcher
+type ListFetcher interface {
```
Comment (donaldthai) [inline pkg/services/fetch.go]: This is good stuff!

File: src/scenes/SystemAdmin/Notifications/NotificationList.jsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import { List, Datagrid, TextField, Filter, TextInput, DateField } from 'react-admin';
+import AdminPagination from 'scenes/SystemAdmin/shared/AdminPagination';
+
+const defaultSort = { field: 'service_member_id', order: 'ASC' };
+
+const NotificationFilter = props => (
+  <Filter {...props}>
```
Comment (donaldthai) [inline src/scenes/SystemAdmin/Notifications/NotificationList.jsx]: Do you wanna add in the filter input styles? This just expands the input field.

```javascript
<Filter {...props} className={styles['system-admin-filters']}>
```

File: pkg/services/fetch/list_fetcher_test.go [hunk 3]
```diff
@@ -69,16 +70,20 @@ func (suite *OfficeUserServiceSuite) TestFetchOfficeUserList() {
...
+		var officeUsers models.OfficeUsers
+		err := fetcher.FetchRecordList(&officeUsers, []services.QueryFilter{}, defaultAssociations(), defaultPagination(), defaultOrdering())
 
 		suite.Error(err)
 		suite.Equal(err.Error(), "Fetch error")
 		suite.Equal(models.OfficeUsers(nil), officeUsers)
 	})
 }
+
+// func (suite *FetchServiceSuite) TestFetchRecordCount() {
```
Comment (donaldthai) [inline pkg/services/fetch/list_fetcher_test.go]: Delete?

File: pkg/services/fetch/list_fetcher.go [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+type listQueryBuilder interface {
+	FetchMany(model interface{}, filters []services.QueryFilter, associations services.QueryAssociations, pagination services.Pagination, ordering services.QueryOrder) error
+	Count(model interface{}, filters []services.QueryFilter) (int, error)
+}
+
+type listFetcher struct {
+	builder listQueryBuilder
+}
+
+// FetchOfficeUserList uses the passed query builder to fetch a list of office users
```
Comment (lynzt) [inline pkg/services/fetch/list_fetcher.go]: modify comment...

File: pkg/services/fetch/list_fetcher.go [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+func (o *listFetcher) FetchRecordList(model interface{}, filters []services.QueryFilter, associations services.QueryAssociations, pagination services.Pagination, ordering services.QueryOrder) error {
+	error := o.builder.FetchMany(model, filters, associations, pagination, ordering)
+	return error
+}
+
+// FetchRecordCount uses the passed query builder to count office users
```
Comment (lynzt) [inline pkg/services/fetch/list_fetcher.go]: does this deal directly w/ office users?

File: pkg/services/fetch/list_fetcher.go [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+func (o *listFetcher) FetchRecordCount(model interface{}, filters []services.QueryFilter) (int, error) {
+	count, error := o.builder.Count(model, filters)
+	return count, error
+}
+
+// NewListFetcher returns an implementation of OfficeUserListFetcher
```
Comment (lynzt) [inline pkg/services/fetch/list_fetcher.go]: does this deal directly w/ office users?

File: pkg/services/fetch/list_fetcher.go (hunk 5 above)
Comment (garrettqmartin8) [reply pkg/services/fetch/list_fetcher.go]: Nope!

File: pkg/services/fetch/list_fetcher.go (hunk 6 above)
Comment (garrettqmartin8) [reply pkg/services/fetch/list_fetcher.go]: Nope!

File: pkg/services/fetch/list_fetcher_test.go (hunk 3 above)
Comment (garrettqmartin8) [reply pkg/services/fetch/list_fetcher_test.go]: Meant to add a test here :)

**Comment by garrettqmartin8** [comment]
@lynzt and @donaldthai I've addressed your comments. Thanks for the thorough review!

**Pull request merged**